app/src/main/java/com/example/music_app/ui/screens/playlists/PlaylistsScreen.kt

```diff
@@ -1,28 +1,116 @@
-import androidx.compose.ui.unit.sp
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.example.music_app.R
+import com.example.music_app.ui.theme.AppTheme
 
+@Preview
 @Composable
-fun PlaylistsScreen() {
+fun PlaylistsScreen(
+    viewModel: PlaylistsViewModel = viewModel(factory = PlaylistsViewModel.Factory)
+) {
+
+    val playlists by viewModel.playlistsForDisplay.collectAsStateWithLifecycle()
+    val isLoading by viewModel.isLoading.collectAsStateWithLifecycle()
+    val lazyListState: LazyListState = rememberLazyListState()
+
+    LaunchedEffect(lazyListState.firstVisibleItemIndex) {
+        viewModel.isScrollOnEnd(lazyListState.firstVisibleItemIndex)
+    }
+
     Column(
-        modifier = Modifier.fillMaxSize(),
-        verticalArrangement = Arrangement.Center,
-        horizontalAlignment = Alignment.CenterHorizontally
+        modifier = Modifier
+            .fillMaxSize()
+            .background(MaterialTheme.colorScheme.background)
+            .padding(AppTheme.dimens.spacing10),
+        verticalArrangement = Arrangement.spacedBy(AppTheme.dimens.spacing10),
+        horizontalAlignment = CenterHorizontally
     ) {
-        Text(
-            text = "Playlists Screen",
-            fontSize = 24.sp,
-            fontWeight = FontWeight.Bold,
-            color = Color.Black
-        )
+        Surface(
+            color = MaterialTheme.colorScheme.primaryContainer,
+            shape = RoundedCornerShape(AppTheme.dimens.spacing08),
+            modifier = Modifier
+                .fillMaxWidth()
+                .height(AppTheme.dimens.spacing80)
+        ) {
+        }
+        Surface(
+            color = MaterialTheme.colorScheme.primaryContainer,
+            shape = RoundedCornerShape(AppTheme.dimens.spacing08),
+            modifier = Modifier
+                .fillMaxWidth()
+                .height(AppTheme.dimens.spacing80)
+        ) {
+        }
+        Box(
+            contentAlignment = Alignment.Center,
+            modifier = Modifier
+                .fillMaxWidth()
+                .weight(1f)
+                .clip(RoundedCornerShape(AppTheme.dimens.spacing08))
+                .background(MaterialTheme.colorScheme.primaryContainer)
+        ) {
+            if (isLoading) {
+                CircularProgressIndicator(
+                    color = Color.Blue,
+                    modifier = Modifier
+                        .height(AppTheme.dimens.spacing80)
+                        .width(AppTheme.dimens.spacing80)
+                )
+            } else {
+                PlaylistsField(
+                    lazyListState = lazyListState,
+                    playlists = playlists,
+                    modifier = Modifier
+                        .fillMaxWidth()
+                )
+            }
+        }
```

Comment:
I think it would be better to move to the separate function not only `LazyColumn` but this piece of code in general